application/src/test/java/com/musicforall/services/feed/FeedServiceTest.java

```diff
@@ -0,0 +1,111 @@
+package com.musicforall.services.feed;
+
+import com.musicforall.dto.feed.Feed;
+import com.musicforall.history.model.History;
+import com.musicforall.history.service.history.HistoryService;
+import com.musicforall.model.Playlist;
+import com.musicforall.model.Track;
+import com.musicforall.model.User;
+import com.musicforall.services.follower.FollowerService;
+import com.musicforall.services.playlist.PlaylistService;
+import com.musicforall.services.track.TrackService;
+import com.musicforall.services.user.UserService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.context.MessageSource;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+import java.util.Map;
+
+import static com.musicforall.history.handlers.events.EventType.PLAYLIST_ADDED;
+import static com.musicforall.history.handlers.events.EventType.TRACK_LIKED;
+import static junit.framework.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author IliaNik on 02.09.2016.
+ */
+
+@RunWith(MockitoJUnitRunner.class)
+public class FeedServiceTest {
+
+    private static final String LOC_1 = "loc1";
+    private static final int TRACK_ID = 3333;
+    private static final int PLAYLIST_ID = 111;
+    private static final int USER1_ID = 2;
+    private static final int USER_ID = 1;
+    private static final String testMessage = "Test message";
+
+    @Mock
+    private UserService userService;
+
+    @Mock
+    private HistoryService historyService;
+
+    @Mock
+    private FollowerService followerService;
+
+    @InjectMocks
+    private FeedServiceImpl feedService;
+
+    @Mock
+    private TrackService trackService;
+
+    @Mock
+    private PlaylistService playlistService;
+
+    @Mock
+    private MessageSource messageSource;
+
+    @Mock
+    private User user;
+
+    @Mock
+    private Track track;
+
+    @Mock
+    private Playlist playlist;
+
+    @Before
+    public void before() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testGetGroupedFollowingFeeds() {
+        final History history1 = new History(TRACK_ID, PLAYLIST_ID,
```

Comment:
You should check every path in swith operator to check that content is generated properly